### PR TITLE
import sozu-acme into the sozu command line

### PR DESCRIPTION
**Pull request by Keksoj**
[sozu-acme](https://github.com/sozu-proxy/sozu-acme) is a configuration tool por Sōzu, that requests Let's Encrypt (or other ACME enabled authorities) to get certificates.

This PR takes the code of sozu acme and embeds it as a sōzu command line option:

```
sozu --config config.toml acme -h    

Set up or renew Let's Encrypt certificates for sozu HTTP reverse proxy

Usage: sozu acme [OPTIONS] --config <CONFIG> --domain <DOMAIN> --email <EMAIL> --id <CLUSTER_ID> --new-cert <NEW_CERTIFICATE_PATH> --chain <CERTIFICATE_CHAIN_PATH> --key-path <KEY_PATH> --http <HTTP_FRONTEND_ADDRESS> --https <HTTPS_FRONTEND_ADDRESS>

Options:
      --config <CONFIG>
          Sets a custom config file
      --domain <DOMAIN>
          the application's domain name
  -t, --timeout <TIMEOUT>
          Sets a custom timeout for commands (in milliseconds). 0 disables the timeout
      --email <EMAIL>
          registration email
      --id <CLUSTER_ID>
          cluster identifier (usually an application)
      --old-cert <OLD_CERTIFICATE_PATH>
          path of the previous certificate (optional)
      --new-cert <NEW_CERTIFICATE_PATH>
          where to write the new certificate
      --chain <CERTIFICATE_CHAIN_PATH>
          where to write the certificate chain
  -k, --key-path <KEY_PATH>
          where to write the key
      --http <HTTP_FRONTEND_ADDRESS>
          HTTP frontend address, in the format IP:port
      --https <HTTPS_FRONTEND_ADDRESS>
          HTTPS frontend address, in the format IP:port
  -h, --help
          Print help information
```

### CODE REVIEW

File: bin/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ include = [
 name = "sozu"
 
 [dependencies]
+acme-lib = "^0.8"
```
Comment (FlorentinDUBOIS) [inline bin/Cargo.toml:28]: Could you write the version entirely?

File: bin/Cargo.toml [hunk 2]
```diff
@@ -48,6 +49,7 @@ rand = "^0.8.5"
 regex = "^1.7.0"
 slab = "^0.4.7"
 smol = "^1.3.0"
+tiny_http = "^0.7"
```
Comment (FlorentinDUBOIS) [inline bin/Cargo.toml:52]: Same here!

File: bin/src/acme.rs [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+/// - contact Let's Encrypt
+/// - retrieve the challenge data
+/// - launch a web server for the HTTP challenge
+/// - configure sōzu to redirect the challenge request to that web server
+/// - start the HTTP challenge validation
+/// - if the challenge was successful, write the certificate, chain and key to the specified paths
+/// - remove the challenge web server from sōzu's configuration
+///
+/// This tool is in beta right now, don't hesitate to test it and report issues.
+pub fn main(
```
Comment (FlorentinDUBOIS) [inline bin/src/acme.rs:38]: I think the name could lead to misunderstanding

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

**Comment by FlorentinDUBOIS** [comment]
Seems good to me, just the version number to keep track, it will allow to put an `=x.x.x` in case something break

File: bin/Cargo.toml (hunk 1 above)
Comment (Keksoj) [reply bin/Cargo.toml]: sure!

**Review by FlorentinDUBOIS — APPROVED** [review]

**Pull request merged**